toolchain/check/testdata/facet/validate_impl_constraints.carbon

```diff
@@ -0,0 +1,44 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// INCLUDE-FILE: toolchain/testing/testdata/min_prelude/facet_types.carbon
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/facet/validate_impl_constraints.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/facet/validate_impl_constraints.carbon
+
+// --- fail_todo_where_impls_tests_associated_constant_of_generic_type.carbon
+
+// C(U) is a type.
+class C(CC:! type) {}
+
+// D(U) impls J and J.X is C(U).
+interface J {
+  let X:! type;
+}
+class D(DD:! type) {}
+final impl forall [U:! type] D(U) as J where .X = C(U) {}
+
+// C(U) impls M if U impls L.
+interface L {}
+interface M {}
+impl forall [U:! L] C(U) as M {}
+
+// V requires that J.X impls M.
+// - J.X must be C(U) => Requires C(U) impls M.
+// - C(U) impls M if U impls L => Requires U impls L.
+fn F(U:! type, V:! type where D(U).(J.X) impls M) {}
```

Comment:
The constraint needs to be written on `U`, not `V`, because it doesn't involve `V`.

Reply:
(See https://docs.carbon-lang.dev/docs/design/generics/details.html#constraints-must-use-a-designator)